### PR TITLE
[ONNX] Part1: update current onnx export 

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- updates current onnx export for further processing in Part2 

Part2 : add slow tests (only local testing) for each model

Any feedback is welcome :hugs:

### CODE REVIEW

File: doctr/models/utils/pytorch.py [hunk 1]
```diff
@@ -116,7 +123,10 @@ def export_classification_model_to_onnx(model: nn.Module, model_name: str, dummy
         input_names=['input'],
         output_names=['logits'],
         dynamic_axes={'input': {0: 'batch_size'}, 'logits': {0: 'batch_size'}},
-        export_params=True, opset_version=13, verbose=False
+        export_params=True,
+        opset_version=14,
```
Comment (frgfm) [inline doctr/models/utils/pytorch.py:127]: Perhaps a comment on the opset_version? Is there a reason to upgrade? (ideally a comment saying why we need at least this version is perfect for future contributors)

File: doctr/models/utils/tensorflow.py [hunk 2]
```diff
@@ -123,33 +122,45 @@ def __repr__(self) -> str:
...
+        kwargs: additional arguments to be passed to tf2onnx
 
     Returns:
         the path to the exported model and a list with the output layer names
     """
+    large_model = kwargs.get('large_model', False)
     model_proto, _ = tf2onnx.convert.from_keras(
         model,
-        opset=13,
+        opset=14,
```
Comment (frgfm) [inline doctr/models/utils/tensorflow.py:153]: same here

**Review by frgfm — COMMENTED** [review]
Thanks :) A few questions in comment

File: doctr/models/utils/tensorflow.py [hunk 3]
```diff
@@ -123,33 +122,45 @@ def __repr__(self) -> str:
...
     Args:
         model: the keras model to be exported
         model_name: the name for the exported model
         dummy_input: the dummy input to the model
+        kwargs: additional arguments to be passed to tf2onnx
 
     Returns:
         the path to the exported model and a list with the output layer names
     """
+    large_model = kwargs.get('large_model', False)
```
[the remark this answers is not in the document]
Comment (felixdittrich92) [reply doctr/models/utils/tensorflow.py:150]: This is a tricky part which has took me a bit time to understood 😅 
In short: while exporting if a weight grow > 2GB it reach the limit in protobuf (only the case with MASTER) so we need to split the model into parts (external tensor storage) these splitted weights are saved in single files and the graph as __PROTO_MODEL.onnx .. all compressed in a zip .. after this onnx is able to load all these related files in one step 

Will be more clear if you see the tests in Part 2 :)

File: doctr/models/utils/pytorch.py (hunk 1 above)
Comment (felixdittrich92) [reply doctr/models/utils/pytorch.py:127]: Mh there is not directly a reason .. we only should keep the latest opset version for future models (higher opset version = more operators supported) :)

File: doctr/models/utils/tensorflow.py (hunk 2 above)
Comment (felixdittrich92) [reply doctr/models/utils/tensorflow.py:153]: Same answer as above 😅

File: doctr/models/utils/pytorch.py (hunk 1 above)
Comment (frgfm) [reply doctr/models/utils/pytorch.py:127]: Alright but then according to https://pytorch.org/docs/master/onnx.html#torch.onnx.export we should match the max opset version of the oldest PyTorch version that we support no?

Comment (felixdittrich92) [reply doctr/models/utils/pytorch.py:127]: Mh.. yes i agree :) 👍

**Comment by felixdittrich92** [comment]
@frgfm after testing again opset 14 is the minimum version we need to ensure all operators are supported (have added a comment) otherwise the export fails for some models in both frameworks 

We should also update torch to a minimum of 1.10 (first support for opset 14 in #791 requirements update)

File: doctr/models/utils/tensorflow.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (frgfm) [reply doctr/models/utils/tensorflow.py:149]: We need to add a comment for this, it's far from obvious haha :)

[the remark this answers is not in the document]
Comment (felixdittrich92) [reply doctr/models/utils/tensorflow.py:149]: @frgfm  now should be ok

**Pull request merged**